jetty-core/jetty-osgi/src/main/java/org/eclipse/jetty/osgi/OSGiDeployableBundleMetadata.java

```diff
@@ -35,27 +30,47 @@
 import org.osgi.framework.Bundle;
 import org.osgi.framework.FrameworkUtil;
 import org.osgi.framework.ServiceRegistration;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import org.osgi.framework.Version;
 
 /**
- * OSGiApp
- *
- * Base class representing info about a WebAppContext/ContextHandler to be deployed into jetty.
+ * Metadata useful for a deployment that will result in a {@link org.eclipse.jetty.server.handler.ContextHandler}
  */
-public class OSGiApp extends App
+public class OSGiDeployableBundleMetadata
 {
-    private static final Logger LOG = LoggerFactory.getLogger(OSGiApp.class);
+    private static final String BUNDLE = OSGiDeployableBundleMetadata.class.getPackageName() + ".bundle";
+    private static final String REGISTRATION = OSGiDeployableBundleMetadata.class.getPackageName() + ".registration";
 
-    protected Bundle _bundle;
-    protected ServiceRegistration _registration;
-    protected ContextHandler _contextHandler;
-    protected String _pathToResourceBase;
-    protected String _contextPath;
-    protected Resource _bundleResource;
+    private final Bundle bundle;
+    private final Path bundlePath;
+    private final String contextPath;
+    private final Properties properties = new Properties();
+    private String pathToResourceBase;
+
+    public OSGiDeployableBundleMetadata(Bundle bundle) throws Exception
+    {
+        this.bundle = bundle;
+        this.bundlePath = getBundlePath(bundle);
+        this.contextPath = getContextPath(bundle);
+    }
+
+    public static void deregisterAsOSGiService(ContextHandler contextHandler)
```

Comment:
This class seems like the wrong place for this method, needs to move either to a utility class or maybe the `AbstractContextProvider`

Reply:
Ditto to above, is `OSGiContextHandlerUtil` ok?

Reply:
There is already a Util class in org.eclipse.jetty.osgi.util.Util - can you add it to that?

Reply:
Done